File: AllCode/Alex_lei/src/quicksort.java
import java.util.ArrayList;
import java.util.Arrays;
import java.util.Scanner;

/**
 * @Author:Alex_lei
 * @Description:
 */
public class quicksort {

    public static void main(String[] args) {
        int[] a = new int[10];
        Scanner sc = new Scanner(System.in);
        for(int i=0;i<10;i++){
            a[i] = sc.nextInt();
        }
        qs(a,0,a.length-1);
        System.out.println(Arrays.toString(a));
    }
    public static void qs(int[] arr,int left,int right){
        int i = left,j = right;
        if(left>right)
            return;
        int key = arr[left];

        while(i!=j){
            while(arr[j]>=key&&i<j){
                j--;
            }
            while(arr[i]<=key&&i<j){
                i++;
            }
            if(i<j){
                int t = arr[i];
                arr[i] = arr[j];
                arr[j] = t;
            }
        }
        arr[left] = arr[i];
        arr[i] = key;
        qs(arr,left,i-1);
        qs(arr,i+1,right);
    }
}

File: AllCode/Scalademo/src/QuickSort.scala
/**
  * @Author:Alex_lei
  * @Description:
  */
object QuickSort {
  var a = List(10,9,8,7,6,5,4,3,2,11)
  var  arr = a.toArray
  def main(args: Array[String]): Unit = {
    quicksort(0,arr.length-1)
    arr.foreach(x=>print(x+" "))
  }

  def quicksort(left:Int,right:Int): Unit = {
    var i = left
    var j = right
    if(left>right){
      return
    }
    val key = arr.apply(left)

    while(i!=j){
      while(arr.apply(j)>=key&&i<j){
        j-=1
      }
      while(arr.apply(i)<=key&&i<j){
        i+=1
      }
      if(i<j){
        val  t = arr.apply(i)
        arr.update(i,arr.apply(j))
        arr.update(j,t)
      }
    }
    arr.update(left,arr.apply(i))
    arr.update(i,key)
    quicksort(left,i-1)
    quicksort(i+1,right)
  }

}
